src/roadmap/common.py

```
@@ -51,6 +51,17 @@ async def decode_header(
     return org_id
 
 
+def _fetch_rbac(fetch_url: str, fetch_headers: dict) -> dict:
+    """Synchronous RBAC fetch function to run in thread pool.
+
+    Takes url and headers as explicit arguments to ensure they are
+    evaluated in the main thread before being passed to the worker thread.
+    """
+    req = urllib.request.Request(fetch_url, headers=fetch_headers)
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)